src/core/streams/stream_message.ts

```diff
@@ -1,40 +1,33 @@
 import { StreamElement } from "../../elements/stream_element"
+import { activateScriptElement, createDocumentFragment } from "../../util"
 
 export class StreamMessage {
   static readonly contentType = "text/vnd.turbo-stream.html"
-  readonly templateElement = document.createElement("template")
+  readonly fragment: DocumentFragment
 
   static wrap(message: StreamMessage | string) {
     if (typeof message == "string") {
-      return new this(message)
+      return new this(createDocumentFragment(message))
     } else {
       return message
     }
   }
 
-  constructor(html: string) {
-    this.templateElement.innerHTML = html
+  constructor(fragment: DocumentFragment) {
+    this.fragment = importStreamElements(fragment)
   }
+}
+
+function importStreamElements(fragment: DocumentFragment): DocumentFragment {
+  for (const element of fragment.querySelectorAll<StreamElement>("turbo-stream")) {
+    const streamElement = document.importNode(element, true)
 
-  get fragment() {
-    const fragment = document.createDocumentFragment()
-    for (const element of this.foreignElements) {
-      fragment.appendChild(document.importNode(element, true))
+    for (const inertScriptElement of streamElement.templateElement.content.querySelectorAll("script")) {
```

Comment:
@seanpdoyle I think this breaks the current behavior we have with the `remove` action https://turbo.hotwired.dev/reference/streams#remove because the [templateElement](https://github.com/hotwired/turbo/blob/main/src/elements/stream_element.ts#L111-L114) requires that the stream element contains a child element `<template>` which for this specific action is not the case

Reply:
Thank you for raising this issue! I've opened https://github.com/hotwired/turbo/pull/665 to resolve it.